### DISCUSSION TITLE
Get exact distance from a ColourDistanceSensor or InfraredSensor

### DISCUSSION THREAD

**Discussion by Bioscreeper** [pybricks/support#1487] [Q&A] [answered]
I'm currently making a robot to move around the room using the Boost hub.
It needs to know the distance to the floor at the front and back, so that it doesn't run into anything, or fall off a table!
However, I can't find what the returned values really mean; the values do not nicely map to centi/millimeters.
I also tried looking into [pybricks-api](https://github.com/pybricks/pybricks-api/blob/master/src/pybricks/pupdevices.py), however it doesn't tell me how it gets the values.
Ideally, I would be able to see the MicroPython code used to evaluate `ir.distance()`

**Comment by laurensvalk** [comment, marked as answer]
It's infrared, so 0% to 100% is the best we can do. The ColorDistance sensor also reports it only in steps of 10%, so the resolution is quite coarse.

The resolution on the infrared sensor is slightly better, but still not something that converts to millimeters.

You'll also find that the "distance" varies depending on the surface material and color. White solid objects will reflect more light, so they appear closer.
